runners/direct-java/src/main/java/org/apache/beam/runners/direct/WriteWithShardingFactory.java

```diff
@@ -78,10 +83,20 @@
     }
   }
 
+
   @Override
   public Map<PValue, ReplacementOutput> mapOutputs(
-      Map<TupleTag<?>, PValue> outputs, PDone newOutput) {
-    return Collections.emptyMap();
+      Map<TupleTag<?>, PValue> outputs, WriteFilesResult newOutput) {
+    // We must connect the new output from WriteFilesResult to the outputs provided by the original
+    // transform.
+    Map.Entry<TupleTag<?>, PValue> original = Iterables.getOnlyElement(outputs.entrySet());
```

Comment:
I think there exists a helper that does all this for you; take a look at other implementations of mapOutputs?

Reply:
Done